<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <!-- Command-line arguments for remote machine, defaults to shrimp opts -->
  <arg name="user" default="shrimp"/>
  <arg name="ip" default="192.168.50.81"/>

  <!-- Load config and parameters -->
  <rosparam file="$(find controller)/configs/$(arg user)_config.yaml" command="load"/>
  <rosparam file="$(find controller)/params.yaml" command="load"/>

  <!-- Launch logger if command-line arg is set -->
  <arg name="log" default="false"/>
  <node if="$(arg log)" name="roslogger" pkg="controller" type="roslogger.py" output="screen"/>

  <!-- Launch joy_node on local machine -->
  <node name="joy" pkg="joy" type="joy_node" output="screen"/>

  <!-- Launch remotely, assuming there is an /env/user_env.sh script -->
  <machine timeout="30" name="lean" address="$(arg ip)" user="$(arg user)" env-loader="/home/$(arg user)/lean_ws/src/controller/env/$(arg user)_env.sh"/>
  <node machine="lean" name="joy_handler" pkg="controller" type="car_joy.py" output="screen"/>
  <node machine="lean" name="sensors" pkg="sensing" type="power_sense.py" output="screen"/>
</launch>
